Mathlib/RingTheory/PowerSeries/Comp.lean

```diff
@@ -0,0 +1,274 @@
+/-
+Copyright (c) 2023 Richard M. Hill. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Richard M. Hill
+-/
+import Mathlib.RingTheory.PowerSeries.Basic
+/-!
+# Definitions
+
+Let `R` be a commutative semiring. Give two formal power series `f(X)` and `g(X)` with coefficients
+in `R`, their formal composition, when it exists, is the power series
+
+  `f ( g ( X ))= ∑ₙ fₙ * g^n`
+
+I.e the `d`-th coefficient of the composition is the sum
+
+  `∑ₙ fₙ * coeff R d (g ^ n)`.
+
+The formal composition exists when all of these sums have finite support. This happens for example
+when `f` is a polynomial, or when the constant term of `g` is nilpotent. There are also other cases
+where the composition is defined, although these cases are less easy to classify.
+
+In this file we define
+
+  `PowerSeries.hasComp` : a relation on `R⟦X⟧`, where `f.hasComp g` means that the formal
+                          composition of `f` and `g` exists.
+
+  `PowerSeries.comp`    : a binary operation on `R⟦X⟧`, where `f.comp g` is the formal composition
+                          in the case `f.hasComp g`, or zero otherwise.
+
```

Comment:
This will be formatted as a code block due to the indent. It would be better as a list:
```suggestion
* `PowerSeries.hasComp` : a relation on `R⟦X⟧`, where `f.hasComp g` means that the formal
  composition of `f` and `g` exists.
* `PowerSeries.comp` : a binary operation on `R⟦X⟧`, where `f.comp g` is the formal composition
  in the case `f.hasComp g`, or zero otherwise.
```

Reply:
The same comment applies below.